src/worker/health.rs

```diff
@@ -8,6 +8,8 @@ use std::{
 use tokio::io::{AsyncBufRead, AsyncWrite};
 use tokio::time::sleep as tokio_sleep;
 
+const CHECK_STATE_INTERVAL_MILLIS: u64 = 100;
+const HEARTBEAT_INTERVAL_SECS: u64 = 5;
```

Comment:
The `Duration` constructors (like `from_millis`) are `const`, so you can call them directly here and remove the unit from the const name :)
```suggestion
const CHECK_STATE_INTERVAL: Duration = Duration::from_millis(100);
const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
```